rpki-validator/src/main/java/net/ripe/rpki/validator3/domain/validation/RpkiRepositoryValidationService.java

```diff
@@ -390,69 +394,67 @@ public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IO
                 super.visitFile(file, attrs);
 
                 final URI objectLocation = currentLocation.resolve(file.getFileName().toString());
+                if (!typePredicate.test(RepositoryObjectType.parse(objectLocation.toString()))) {
+                    return FileVisitResult.CONTINUE;
+                }
+
                 validationResult.setLocation(new ValidationLocation(objectLocation));
 
                 final long objectSize = file.toFile().length();
                 if (objectSize > RpkiObject.MAX_SIZE) {
                     validationResult.error(ErrorCodes.REPOSITORY_OBJECT_MAXIMUM_SIZE, objectLocation.toASCIIString(), String.valueOf(objectSize), String.valueOf(RpkiObject.MAX_SIZE));
                 } else {
-                    final byte[] content = Files.readAllBytes(file);
-                    final byte[] sha256 = Sha256.hash(content);
-
-                    final String key = Hex.format(sha256);
                     final String location = validationResult.getCurrentLocation().getName();
-
-                    boolean exists = existingObjectsKeys.contains(key);
-
-                    if (!exists) {
-                        if (workCounter.get() > threshold) {
-                            storeObject(tx, validationRun, existingObjectsKeys);
-                            workCounter.decrementAndGet();
-                        }
-                        asyncCreateObjects.submit(() -> RpkiObjectUtils.createRpkiObject(location, content));
-                        workCounter.incrementAndGet();
-                    } else {
-                        rpkiObjects.addLocation(tx, Key.of(key), location);
+                    final byte[] content = Files.readAllBytes(file);
+                    pendingObjects.add(Pair.of(location, content));
```

Comment:
Since `<uri, blob>` is kind of common in the data model of the validator, it may be worth it to use a value class?